### PR TITLE
Doc update for LineSplitter 

### PR DESCRIPTION
**Pull request by jpsiitonen**
LineSplitter class documentation improved with a example code

### CODE REVIEW

File: sdk/lib/convert/line_splitter.dart [hunk 1]
```diff
@@ -10,12 +10,31 @@ const int _CR = 13;
...
+/// * a final non-empty line can be ended by the end of the string
 ///
 /// The returned lines do not contain the line terminators.
-
+///
+/// Example:
+/// ```dart
+/// const LineSplitter ls = LineSplitter();
+/// const String sampleText =
+///     'Dart is: \ran object-oriented \nclass-based \ngarbage-collected'
```
Comment (alexlindroos) [inline sdk/lib/convert/line_splitter.dart]: What is \ran

Comment (tomgilder) [reply sdk/lib/convert/line_splitter.dart]: It's a carriage return and "an"

Comment (jpsiitonen) [reply sdk/lib/convert/line_splitter.dart]: Yes. I think it is better to change it to:
```
const String sampleText =
    'Dart is: \r an object-oriented \n class-based \n garbage-collected'
    '\r\n language with C-style syntax \r\n';
```
    
Even though output contains then whitespaces but it is easier to spot \r and \n also on html page. 

Output: 
```
0: Dart is: 
1:  an object-oriented 
2:  class-based 
3:  garbage-collected 
4:  language with C-style syntax
```

Comment (alexlindroos) [reply sdk/lib/convert/line_splitter.dart]: oooooh I see

File: sdk/lib/convert/line_splitter.dart [hunk 2]
```diff
@@ -22,18 +22,18 @@ const int _CR = 13;
...
 /// }
 /// // 0: Dart is:
-/// // 1: an object-oriented
-/// // 2: class-based
-/// // 3: garbage-collected
-/// // 4: language with C-style syntax
+/// // 1:  object-oriented
+/// // 2:  class-based
+/// // 3:  garbage-collected
+/// // 4:  language with C-style syntax
```
Comment (tomgilder) [inline sdk/lib/convert/line_splitter.dart:36]: This is not the output of the sample. "An" is missing

**Review by tomgilder — CHANGES_REQUESTED** [review]

File: sdk/lib/convert/line_splitter.dart (hunk 2 above)
Comment (jpsiitonen) [reply sdk/lib/convert/line_splitter.dart:36]: oh dear..

**Review by tomgilder — APPROVED** [review]

**Review by alexlindroos — APPROVED** [review]

**Review by stepthomp — APPROVED** [review]

**Pull request closed without merging**